DatadogInternal/Sources/Utils/UIKitExtensions.swift

```diff
@@ -0,0 +1,24 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+#if canImport(UIKit)
+import UIKit
+
+extension DatadogExtension where ExtendedType == UIApplication {
+    /// `UIApplication.shared` does not compile in some environments (e.g. notification service app extension), resulting with:
+    /// _"shared' is unavailable in application extensions for iOS: Use view controller based solutions where appropriate instead"_.
+    ///
+    /// As a workaround, this `managedShared` utility provides a key-path access to the `UIApplication.shared` to make the compiler pass.
+    public static var managedShared: UIApplication? {
```

Comment:
The property is now accessed using the `DatadogExtended`:
```swift
UIApplication.dd.managedShared
```

Reply:
❓ 
How would customers use this API?

Reply:
This extension is for internal use only, not for customer. That's why it is expose using the `DatadogExtended` protocol as part of the `DatadogInternal` module. The `. managedShared` is used by both the core and the RUM modules.